src/main/java/com/ogp/configurator/ConfigService.java

```diff
@@ -63,61 +66,73 @@ private ConfigService(Builder builder) {
-					case NODE_UPDATED:
-						if (configEntityClass != null 
-										&& isInitialized) {
-							Object obj = JacksonSerializator.Deserialize(event.getData().getData(), configEntityClass);
-							String key = childDataToKey(event.getData());
-							Map<String, Object> entities = configObjects.putIfAbsent(configEntityClass, new HashMap<String, Object>());
-							synchronized (entities) {
-								entities.put(key, obj);
-							}							
-							System.out.printf("childEvent() key=%s put in cache, now %d classes and %d objects\n", 
-									key,
-									configObjects.size(),
-									configObjects.get(configEntityClass).size());
-						}
-						break;
-					case NODE_REMOVED:
-						if (configEntityClass != null
-										&& isInitialized) {
-							String key = childDataToKey(event.getData());
-							if (configObjects.containsKey(configEntityClass)) {
-								Map<String, Object> entities = configObjects.get(configEntityClass);
-								synchronized (entities) {
-									entities.remove(key);
-								}
-							}
-						}
-						break;
-					case INITIALIZED:
-						reloadConfigTree();
-						System.out.println("INITIALIZED, COMPLETE");
-						isInitialized = true;
-						isReadOnly = false;
-						break;
-					case CONNECTION_LOST:
-					case CONNECTION_SUSPENDED:
-						isReadOnly = true;
-						break;
-					case CONNECTION_RECONNECTED:
-						isReadOnly = false;
-						break;
-				}
+				logger.debug("== Config event: type=" + event.getType() + " childData={" + childDataToString(event.getData()) + "}");
+				processEvent(event);
 			}
 		});
 	}
 
+	private void processEvent(TreeCacheEvent event) throws Exception {
+		Class<Object> configEntityClass = childDataToClass(event.getData());
+		switch (event.getType()) {
+			case NODE_ADDED:
+			case NODE_UPDATED:
+				if (configEntityClass != null 
+								&& isInitialized) {
+					Object obj = serializer.Deserialize(event.getData().getData(), configEntityClass);
+					String key = childDataToKey(event.getData());
+					Map<String, Object> entities = configObjects.putIfAbsent(configEntityClass, new HashMap<String, Object>());
+					synchronized (entities) {
+						entities.put(key, obj);
+					}							
+					logger.trace("key=%s put/update in cache, now %d classes and %d objects of this key in cache\n", 
+							key,
+							configObjects.size(),
+							configObjects.get(configEntityClass).size());
+				}
+				break;
+			case NODE_REMOVED:
+				if (configEntityClass != null
+								&& isInitialized) {
+					String key = childDataToKey(event.getData());
+					if (configObjects.containsKey(configEntityClass)) {
+						Map<String, Object> entities = configObjects.get(configEntityClass);
+						synchronized (entities) {
+							entities.remove(key);
+						}
+					}
+				}
+				break;
+			case INITIALIZED:
+				reloadConfigTree();
+				logger.info("Initialization complete");
+				isInitialized = true;
+				isReadOnly = false;
+				synchronized (syncReady) {
+					syncReady.notify();
```

Comment:
Lets not to use Java low level synchronization primitives, but more high level utils from java concurrency (e.g. count down latch)